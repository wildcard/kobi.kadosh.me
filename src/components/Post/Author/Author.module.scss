@use "../../../assets/scss/variables" as vars;
@use "../../../assets/scss/mixins" as mix;

.author {
  border-top: 1px solid vars.$color-gray-border;
  max-width: vars.$layout-post-width;
  padding-top: 20px;
  @include mix.line-height(1);
  @include mix.margin-top(1);
  @include mix.margin-bottom(2);

  &__bio {
    &-twitter {
      display: block;
      text-decoration: underline;
    }
  }
}

@include mix.breakpoint-sm {
  .author {
    margin-left: auto;
    margin-right: auto;
  }

}

:export {
  author: author;
  author__bio: author__bio;
  author__bio-twitter: author__bio-twitter;
}