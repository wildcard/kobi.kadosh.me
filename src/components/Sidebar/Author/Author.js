// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import Img from 'gatsby-image';
import type { FluidObject } from 'gatsby-image';
import styles from './Author.module.scss';

type Props = {
  author: {
    name: string,
    bio: string
  },
  photo: {
      childImageSharp: {
        fluid: FluidObject
      }
    },
  isIndex: ?boolean,
};

const Author = ({ author, photo, isIndex }: Props) => (
  <div className={styles['author']}>
    <Link to="/">
      <Img
        fluid={photo.childImageSharp.fluid}
        className={styles['author__photo']}
        width="75"
        height="75"
        alt={author.name}
      />
    </Link>

    { isIndex === true ? (
      <h1 className={styles['author__title']}>
        <Link className={styles['author__title-link']} to="/">
          {author.name}
        </Link>
      </h1>
    ) : (
      <h2 className={styles['author__title']}>
        <Link className={styles['author__title-link']} to="/">
          {author.name}
        </Link>
      </h2>
    )}
    <p className={styles['author__subtitle']}>{author.bio}</p>
  </div>
);

export default Author;
