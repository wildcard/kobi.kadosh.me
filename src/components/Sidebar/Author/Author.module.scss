@use "../../../assets/scss/variables" as vars;
@use "../../../assets/scss/mixins" as mix;

.author {
  &__photo {
      width: 75px;
      height: 75px;
    margin-bottom: 0;
    border-radius: 50%;
    background-clip: padding-box
  }

  &__title {
    font-size: vars.$typographic-base-font-size * 1.125;
    font-weight: 600;
    @include mix.line-height(1.125);
    @include mix.margin(.5, 0, .5, 0);

    &-link {
      color: vars.$color-base;

      &:hover,
      &:focus {
        color: vars.$color-base
      }

    }

  }

  &__subtitle {
    color: vars.$color-gray;
    @include mix.line-height(1);
    @include mix.margin-bottom(1)
  }

}


:export {
  author: author;
  author__photo: author__photo;
  author__title: author__title;
  author__title-link: author__title-link;
  author__subtitle: author__subtitle;
}