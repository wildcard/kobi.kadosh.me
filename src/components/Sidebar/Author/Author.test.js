// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Author from './Author';

describe('Author', () => {
  const props = {
    author: {
      name: 'test',
      bio: 'test',
    },
    photo: {
      childImageSharp: {
        fluid: {
          base64: 'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAAB2YR5xbFVKoukE//EAB0QAQACAQUBAAAAAAAAAAAAAAIAAwEEEBEhIjP/2gAIAQEAAQUCsWSKkkJlczT9CX/Iedv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAwEBPwEh/8QAFREBAQAAAAAAAAAAAAAAAAAAEAH/2gAIAQIBAT8BKf/EABsQAQACAgMAAAAAAAAAAAAAAAEQEQAhAiJh/9oACAEBAAY/AlC3L5APkdrDK3Oo/8QAHBAAAgICAwAAAAAAAAAAAAAAAREAITFREEGx/9oACAEBAAE/IUPLAliQVq4IFCqjvdmHqOHYbmwRj2Of/9oADAMBAAIAAwAAABAvCAD/xAAXEQADAQAAAAAAAAAAAAAAAAABECEx/9oACAEDAQE/EIFoL//EABcRAQADAAAAAAAAAAAAAAAAABABITH/2gAIAQIBAT8QizB//8QAHRABAQACAgMBAAAAAAAAAAAAAREAITFBEFHBsf/aAAgBAQABPxBm4eWVs+5NAPYUe978WugFJNcuFD10O1wh1jhaolDT0L8xKhE7Pofjirn/2Q==',
          aspectRatio: 1,
          src: '/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg',
          srcSet: '/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/f3ac7/kobi-kadosh-thumb-256.jpg 32w,\n/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/116b2/kobi-kadosh-thumb-256.jpg 64w,\n/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg 128w,\n/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/97817/kobi-kadosh-thumb-256.jpg 192w,\n/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/c296b/kobi-kadosh-thumb-256.jpg 256w',
          sizes: '(max-width: 128px) 100vw, 128px'
        }
      }
    },
    isIndex: false
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Author {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
