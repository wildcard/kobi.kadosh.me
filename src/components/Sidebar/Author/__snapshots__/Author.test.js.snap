// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div
  className="author"
>
  <a
    href="/"
  >
    <div
      className="author__photo gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "100%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAIEBQP/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAAB2YR5xbFVKoukE//EAB0QAQACAQUBAAAAAAAAAAAAAAIAAwEEEBEhIjP/2gAIAQEAAQUCsWSKkkJlczT9CX/Iedv/xAAVEQEBAAAAAAAAAAAAAAAAAAAQAf/aAAgBAwEBPwEh/8QAFREBAQAAAAAAAAAAAAAAAAAAEAH/2gAIAQIBAT8BKf/EABsQAQACAgMAAAAAAAAAAAAAAAEQEQAhAiJh/9oACAEBAAY/AlC3L5APkdrDK3Oo/8QAHBAAAgICAwAAAAAAAAAAAAAAAREAITFREEGx/9oACAEBAAE/IUPLAliQVq4IFCqjvdmHqOHYbmwRj2Of/9oADAMBAAIAAwAAABAvCAD/xAAXEQADAQAAAAAAAAAAAAAAAAABECEx/9oACAEDAQE/EIFoL//EABcRAQADAAAAAAAAAAAAAAAAABABITH/2gAIAQIBAT8QizB//8QAHRABAQACAgMBAAAAAAAAAAAAAREAITFBEFHBsf/aAAgBAQABPxBm4eWVs+5NAPYUe978WugFJNcuFD10O1wh1jhaolDT0L8xKhE7Pofjirn/2Q=="
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 128px) 100vw, 128px"
          srcSet="/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/f3ac7/kobi-kadosh-thumb-256.jpg 32w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/116b2/kobi-kadosh-thumb-256.jpg 64w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg 128w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/97817/kobi-kadosh-thumb-256.jpg 192w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/c296b/kobi-kadosh-thumb-256.jpg 256w"
        />
        <img
          alt="test"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 128px) 100vw, 128px"
          src="/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg"
          srcSet="/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/f3ac7/kobi-kadosh-thumb-256.jpg 32w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/116b2/kobi-kadosh-thumb-256.jpg 64w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg 128w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/97817/kobi-kadosh-thumb-256.jpg 192w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/c296b/kobi-kadosh-thumb-256.jpg 256w"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/f3ac7/kobi-kadosh-thumb-256.jpg 32w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/116b2/kobi-kadosh-thumb-256.jpg 64w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg 128w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/97817/kobi-kadosh-thumb-256.jpg 192w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/c296b/kobi-kadosh-thumb-256.jpg 256w\\" sizes=\\"(max-width: 128px) 100vw, 128px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 128px) 100vw, 128px\\" srcset=\\"/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/f3ac7/kobi-kadosh-thumb-256.jpg 32w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/116b2/kobi-kadosh-thumb-256.jpg 64w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg 128w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/97817/kobi-kadosh-thumb-256.jpg 192w,
/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/c296b/kobi-kadosh-thumb-256.jpg 256w\\" src=\\"/static/f2f503f5b0e8c840dc4a02e3edbdb3b7/7c5ac/kobi-kadosh-thumb-256.jpg\\" alt=\\"test\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
  <h2
    className="author__title"
  >
    <a
      className="author__title-link"
      href="/"
    >
      test
    </a>
  </h2>
  <p
    className="author__subtitle"
  >
    test
  </p>
</div>
`;
